### ISSUE TITLE
Random commas appear in random places in the output JavaScript

### ISSUE THREAD

**Issue by TriEdgeAI** [bridgedotnet/Bridge#1412]
### Expected
```javascript
    Bridge.define('Test.BridgeIssues.N1412.SimpleTimeScaleController', {
        getComponent: function (T) {
            return null;
        },
        updateInternal: function () {
            var $t;
            // There should be a teml JS variavble generated with no comma
            var animationComp = this.getComponent(String);
    
            if (animationComp != null) {
                $t = Bridge.getEnumerator(animationComp);
                while ($t.moveNext()) {
                    var state = $t.getCurrent();
    
                }
            }
        }
    });
```

### Actual
```javascript
Bridge.define('Demo.SimpleTimeScaleController', {
    getComponent: function (T) {
        return null;
    },
    updateInternal: function () {
        var , $t;
        // There should be a teml JS variavble generated with no comma
        var animationComp = this.getComponent(String);

        if (animationComp != null) {
            $t = Bridge.getEnumerator(animationComp);
            while ($t.moveNext()) {
                var state = $t.getCurrent();

            }
        }
    }
});
```

### Steps To Reproduce
Example([Live](http://live.bridge.net/#ec224a38f0485a0e3b0b23ca913c7ffa)):
```csharp
public class SimpleTimeScaleController
{
    public List<string> GetComponent<T>()
    {
        return null;
    }

    private void UpdateInternal()
    {
        // There should be a teml JS variavble generated with no comma
        var animationComp = GetComponent<string>();

        if (animationComp != null)
        {
            foreach (string state in animationComp)
            {

            }
        }
    }
}
```

I also made a few attempts to change the code piece a little and see if I can get it to stop outputting the comma. If I replace `GetComponent<string>()` with `null` it disappears.

**Comment by vladsch** [comment]
Thanks for the report
Short test case
```csharp
public class SimpleTimeScaleController
    {
        public List<string> GetComponent<T>()
        {
            return null;
        }

        private void UpdateInternal()
        {
            var animationComp = GetComponent<string>();

            if (animationComp != null)
            {
                foreach (string state in animationComp)
                {
                   
                }
            }
        }
    }
```

**Comment by LeonidVeriga** [comment]
Hi @TriEdgeAI, we are going to include the fix into `1.14` release.

**Comment by LeonidVeriga** [comment]
Updated [initial description](https://github.com/bridgedotnet/Bridge/issues/1412#issue-157338339) with the sample narrowed down and Bridge Live sample
